### PR TITLE
test: e2e tests for replication slots

### PR DESCRIPTION
**Pull request by wadlejitendra**

### CODE REVIEW

File: tests/e2e/fastfailover_test.go [hunk 1]
```diff
@@ -87,6 +91,10 @@ var _ = Describe("Fast failover", Serial, Label(tests.LabelPerformance), func()
 	})
 
 	Context("with async replicas cluster", func() {
+		if strings.Contains(os.Getenv("POSTGRES_IMG"), ":10") {
+			// Cluster file without replication slot since it requires PostgreSQL 11 or above
+			sampleFile = sampleFileWithoutReplSlots
+		}
```
Comment (mnencia) [inline tests/e2e/fastfailover_test.go:97]: I would prefer to have the PostgreSQL version in the env structure, so we can detect it in a centralized and controllable way.

File: tests/utils/replication_slots.go [hunk 2]
```diff
@@ -80,10 +80,14 @@ func GetRepSlotsOnPod(namespace, podName string, env *TestingEnvironment) ([]str
...
 		return nil, err
 	}
-
-	slots := strings.Split(strings.TrimSpace(stdout), "\n")
-	sort.Strings(slots)
-	return slots, err
+	var slots []string
+	// To avoid list with space entry when stdout value is empty
+	// then just skip split and return empty list.
+	if !(stdout == "") {
```
Comment (mnencia) [inline tests/utils/replication_slots.go:86]:
```suggestion
	if stdout != "" {
```

**Review by mnencia — CHANGES_REQUESTED** [review]

File: tests/e2e/fastfailover_test.go (hunk 1 above)
Comment (danishedb) [reply tests/e2e/fastfailover_test.go]: Resolved.

File: tests/utils/replication_slots.go (hunk 2 above)
Comment (danishedb) [reply tests/utils/replication_slots.go]: Resolved.

File: tests/utils/environment.go [hunk 3]
```diff
@@ -71,19 +72,30 @@ type TestingEnvironment struct {
...
-	var err error
+	// Fetching postgres image version.
+	if postgresImage, exist := os.LookupEnv("POSTGRES_IMG"); exist {
+		imageReference := utils.NewReference(postgresImage)
+		postgresImageVersion, err := postgres.GetPostgresVersionFromTag(imageReference.Tag)
+		if err != nil {
+			return nil, err
+		}
+		env.PostgresVersion = postgresImageVersion / 10000
+	}
```
Comment (mnencia) [inline tests/utils/environment.go:97]: What happens if the POSTGRES_IMG variable is not defined? I believe the tests run on the default version, so we should set `env.PostgresVersion` accordingly.

Comment (danishedb) [reply tests/utils/environment.go]: Added it now.

File: tests/e2e/asserts_test.go [hunk 4]
```diff
@@ -2501,44 +2501,38 @@ func AssertRepSlotsOnPod(
 	clusterName string,
 	pod corev1.Pod,
 ) {
+	// Replication slot high availability requires PostgreSQL 11 or above
+	if env.PostgresVersion == 10 {
+		GinkgoWriter.Printf("Ignoring replication slots verification for postgres 10")
```
Comment (jsilvela) [inline tests/e2e/asserts_test.go:2506]: this will mark the test as succeeded. Should skip...

File: tests/e2e/scaling_test.go [hunk 5]
```diff
@@ -28,11 +28,13 @@ import (
...
-		sampleFile  = fixturesDir + "/base/cluster-storage-class.yaml.template"
-		clusterName = "postgresql-storage-class"
-		level       = tests.Lowest
+		namespace                 = "cluster-scale-e2e-storage-class"
+		sampleFileWithoutRepSlots = fixturesDir + "/base/cluster-storage-class.yaml.template"
+		sampleFileWithRepSlots    = fixturesDir + "/base/cluster-storage-class-with-rep-slots.yaml.template"
+		clusterName               = "postgresql-storage-class"
+		level                     = tests.Lowest
 	)
+	sampleFile := sampleFileWithRepSlots
```
Comment (jsilvela) [inline tests/e2e/scaling_test.go:37]: yikes. We're changing this test, as well as the switchover / failover tests.
And no longer test without replication slots (except in postgres 10).
We should add new tests, not subtract!!

**Review by jsilvela — APPROVED** [review]

**Review by NiccoloFei — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

**Pull request merged**